### ISSUE TITLE
Inclusive or exclusive range filters in API

### ISSUE THREAD

**Issue by mik-laj** [apache/airflow#9237]
Helllo,

We have the following ranger filter parameters:

- FilterDurationGTE
- FilterDurationLTE
- FilterEndDateGTE
- FilterEndDateLTE
- FilterExecutionDateGTE
- FilterExecutionDateLTE
- FilterStartDateGTE
- FilterStartDateLTE

Only closed ranged(in interval notation: [start_xxx, end_xxx]) can be fetched using them.

I think, filters representing ranges should use inclusive start values and exclusive end values (half-closed intervals); in interval notation: [start_xxx, end_xxx).

Exclusive end values are preferable for the following reasons:

* It conforms to user expectations, particularly for continuous values such as timestamps, and avoids the need to express imprecise “limit values” (e.g. 2012-04-20T23:59:59).
* It is consistent with most common programming languages, including C++, Java, Python, and Go.
* It is easier to reason about abutting ranges: [0, x), [x, y), [y, z), where values are chainable from one range to the next.

More information: https://google.aip.dev/145

Alternatively, we can also add LT and GT filters, but this could only complicate situations without much benefit. We would have 4 different parameters for one field. Changing parameter names and changing behavior seems better to me.

Best regards,
Kamil Breguła

**Comment by mik-laj** [comment]
@houqp I haven't started the discussion on the mailing list about this problem yet, but maybe you have some thought?

**Comment by houqp** [comment]
@mik-laj yes, ` [start_xxx, end_xxx)` is much easier to use than ` [start_xxx, end_xxx]` from a client's point of view. If we don't want to support all 4 comparison operators, then I would love to see us at least switch to GTE and LT.

**Comment by potiuk** [comment]
Yep. Agree using GTE and LT is a better approach. I think it also encourages good patterns.

However I think this is only valid in case of dates, not durations. For duration i would also like  to have <= 200 seconds not only >= 200 seconds (but it also depends on the resolution of duration).  

So I'd opt for all 4 operators for duration but only GTE/LT for dates.

**Comment by Bowrna** [comment]
Any update on the question that I have posted? Gentle reminder..

**Comment by potiuk** [comment]
I think we should simply add GT and LT filters everywhere. It is simplest and let the user choose what filter they want.

**Comment by Bowrna** [comment]
ok @potiuk am i right to assume that this issue is about removing LTE and GTE and keeping LT and GT filters alone?

**Comment by potiuk** [comment]
I think we should keep both.

**Comment by uranusjr** [comment]
This shouldn’t be too difficult. Just add GT and LT variants of the GTE/LTE filters listed above, and add them as parameters that make sense in the API spec (`openapi/v1.yaml`). Feel free to contribute if you are inclined to.

**Comment by kirubankamaraj** [comment]
@Bowrna @uranusjr I'm interested to support GT and LT, can you assign this to me?

**Comment by kirubankamaraj** [comment]
@mik-laj  @uranusjr Changes merged, Kindly close this issue.
